src/components/AppLayout/Menu.tsx

```diff
@@ -50,34 +51,39 @@ interface Props {
 	className?: string;
 }
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const Menu: FC<Props> = ({ className }) => {
 	const { multisigAddresses, activeMultisig, setUserDetailsContextState } = useGlobalUserDetailsContext();
+	const { network } = useGlobalApiContext();
 	const [selectedMultisigAddress, setSelectedMultisigAddress] = useState('');
 	const { openModal } = useModalContext();
 	const location = useLocation();
 
 	useEffect(() => {
-		if(activeMultisig){
+
+		if(multisigAddresses.filter((multisig) => multisig.network === network).find((multisig) => multisig.address === activeMultisig)){
 			setSelectedMultisigAddress(activeMultisig);
 		}
-		else if(multisigAddresses && multisigAddresses[0]){
-			setSelectedMultisigAddress(multisigAddresses[0].address);
-			setUserDetailsContextState(prevState => {
-				return {
-					...prevState,
-					activeMultisig: multisigAddresses[0].address
-				};
-			});
-		}
 		else{
-			setSelectedMultisigAddress('');
+			if((multisigAddresses.filter((multisig) => multisig.network === network).length )){
```

Comment:
store multisigAddresses.filter((multisig) => multisig.network === network) in a const